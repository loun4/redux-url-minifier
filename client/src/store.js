
import { createStore, applyMiddleware, combineReducers } from 'redux';
import thunkMiddleware from 'redux-thunk';
import { createLogger } from 'redux-logger';
import { createForms } from 'react-redux-form';
import entities from './reducers/entities';

const loggerMiddleware = createLogger({
  predicate() { return process.env.NODE_ENV === 'development'; },
});

const forms = createForms({
  linkForm: {
    linkURL: '',
  },
  signinForm: {
    login: '',
    password: '',
  },
});

export default function configureStore(preloadedState) {
  return createStore(
    combineReducers({
      ...forms,
      entities,
    }),
    preloadedState,
    applyMiddleware(
      thunkMiddleware,
      loggerMiddleware,
    ),
  );
}
