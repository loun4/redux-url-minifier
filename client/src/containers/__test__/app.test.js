
import React from 'react';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import RouterProvider from '../router-provider';
import store from '../../store';
import App from '../app';

let subject;
const testStore = store();

const buildSubject = props => (
  <RouterProvider>
    <Provider store={testStore}>
      <App {...props}>
        <div />
      </App>
    </Provider>
  </RouterProvider>
);

beforeEach(() => {
  subject = buildSubject();
});

test('Render app', () => {
  const component = renderer.create(subject);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Render error on get', () => {
  subject = buildSubject({
    error: {
      type: 'get',
    },
  });

  const component = renderer.create(subject);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Render error on save', () => {
  subject = buildSubject({
    error: {
      type: 'save',
    },
  });

  const component = renderer.create(subject);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test.only('Call authenticate on componentDidMount if session is ready for auth', () => {
  const authenticate = jest.fn();

  subject = buildSubject({
    session: {
      ...testStore.getState().session,
      readyToAuthenticate: true,
    },
    authenticate,
  });

  mount(subject);
  expect(authenticate).toHaveBeenCalled();
});
