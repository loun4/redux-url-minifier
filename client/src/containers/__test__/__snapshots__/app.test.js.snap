// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render app 1`] = `
<div>
  <div
    className="ui inverted menu"
  >
    <a
      className="header item active"
      href="/"
      onClick={[Function]}
    >
      URL shortener
    </a>
    <div
      className="right menu"
    >
      <a
        className="item"
        href="/admin"
        onClick={[Function]}
      >
        Admin
      </a>
    </div>
  </div>
  <div
    className="app ui container"
  >
    <div />
    <div
      className="ui negative message notification"
    >
      Something worng happened, please try again.
    </div>
  </div>
</div>
`;

exports[`Render error on get 1`] = `
<div
  className="app ui container"
>
  <div
    className="ui negative message massive error-box"
  >
    Something worng happened, please try again.
  </div>
</div>
`;

exports[`Render error on save 1`] = `
<div>
  <div
    className="ui inverted menu"
  >
    <a
      className="header item active"
      href="/"
      onClick={[Function]}
    >
      URL shortener
    </a>
    <div
      className="right menu"
    >
      <a
        className="item"
        href="/admin"
        onClick={[Function]}
      >
        Admin
      </a>
    </div>
  </div>
  <div
    className="app ui container"
  >
    <div />
    <div
      className="ui negative message notification is-visible"
    >
      Something worng happened, please try again.
    </div>
  </div>
</div>
`;
